dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/ClaimedTaskRestRepository.java

```diff
@@ -0,0 +1,163 @@
+
+    @Override
+    public ClaimedTaskRest findOne(Context context, Integer id) {
+        ClaimedTask task = null;
+        try {
+            task = claimedTaskService.find(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        if (task == null) {
+            return null;
+        }
+        return converter.fromModel(task);
+    }
+
+    @SearchRestMethod(name = "findByUser")
+    public Page<ClaimedTaskRest> findByUser(@Param(value = "uuid") UUID userID, Pageable pageable) {
+        List<ClaimedTask> tasks = null;
+        try {
+            Context context = obtainContext();
+            EPerson ep = epersonService.find(context, userID);
+            tasks = claimedTaskService.findByEperson(context, ep);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Page<ClaimedTaskRest> page = utils.getPage(tasks, pageable).map(converter);
+        return page;
+    }
+
+    @Override
+    public Class<ClaimedTaskRest> getDomainClass() {
+        return ClaimedTaskRest.class;
+    }
+
+    @Override
+    public ClaimedTaskResource wrapResource(ClaimedTaskRest task, String... rels) {
+        return new ClaimedTaskResource(task, utils, rels);
+    }
+
+    @Override
+    protected ClaimedTaskRest action(Context context, HttpServletRequest request, Integer id)
+        throws SQLException, IOException, AuthorizeException {
+        ClaimedTask task = null;
+        task = claimedTaskService.find(context, id);
+        XmlWorkflowServiceFactory factory = (XmlWorkflowServiceFactory) XmlWorkflowServiceFactory.getInstance();
+        Workflow workflow;
+        try {
+            workflow = factory.getWorkflowFactory().getWorkflow(task.getWorkflowItem().getCollection());
+
+            Step step = workflow.getStep(task.getStepID());
+            WorkflowActionConfig currentActionConfig = step.getActionConfig(task.getActionID());
+            workflowService
+                .doState(context, context.getCurrentUser(), request, task.getWorkflowItem().getID(), workflow,
+                    currentActionConfig);
+            if (!Action.getErrorFields(request).isEmpty()) {
+                throw new UnprocessableEntityException("Missing required fields");
+            }
+            // workflowRequirementsService.removeClaimedUser(context, task.getWorkflowItem(), task.getOwner(), task
+            // .getStepID());
+            context.addEvent(new Event(Event.MODIFY, Constants.ITEM, task.getWorkflowItem().getItem().getID(), null,
+                itemService.getIdentifiers(context, task.getWorkflowItem().getItem())));
+        } catch (WorkflowConfigurationException | MessagingException | WorkflowException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    @Override
+    protected void delete(Context context, Integer id) {
```

Comment:
Is this supposed to return the task to the pool (no longer claimed)?

Reply:
yes
https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#delete-method

Reply:
it is actually deleting only the claimed task. Since I see that it leads to discussion it is obviously not self-explaining so I have added javadoc anyway